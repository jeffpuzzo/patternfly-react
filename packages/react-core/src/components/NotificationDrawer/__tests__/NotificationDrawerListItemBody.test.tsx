import React from 'react';
import { render, screen } from '@testing-library/react';
import { NotificationDrawerListItemBody } from '../NotificationDrawerListItemBody';

describe('NotificationDrawerListItemBody', () => {
  test('renders with PatternFly Core styles', () => {
    render(<NotificationDrawerListItemBody data-testid="test-id" />);
    expect(screen.getByTestId('test-id').outerHTML).toMatchSnapshot();
  });

  test('className is added to the root element', () => {
    render(<NotificationDrawerListItemBody className="extra-class" data-testid="test-id" />);
    expect(screen.getByTestId('test-id').className).toMatchSnapshot();
  });

  test('list item body with timestamp property applied', () => {
    render(<NotificationDrawerListItemBody timestamp="5 minutes ago" data-testid="test-id" />);
    expect(screen.getByTestId('test-id').className).toMatchSnapshot();
  });
});
