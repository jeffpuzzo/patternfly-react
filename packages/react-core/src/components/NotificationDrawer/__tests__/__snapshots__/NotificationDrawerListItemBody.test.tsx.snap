// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationDrawerListItemBody className is added to the root element 1`] = `"pf-c-notification-drawer__list-item-description extra-class"`;

exports[`NotificationDrawerListItemBody list item body with timestamp property applied 1`] = `"pf-c-notification-drawer__list-item-description"`;

exports[`NotificationDrawerListItemBody renders with PatternFly Core styles 1`] = `"<div data-testid=\\"test-id\\" class=\\"pf-c-notification-drawer__list-item-description\\"></div>"`;
